### DISCUSSION TITLE
Filter data occuring in multiple datasets

### DISCUSSION THREAD

**Discussion by JulesVandenbroeck** [columnflow/columnflow#363] [Q&A]
Hi everyone,

Currently the events from data are not filtered if they occur multiple times in the analysis (for example I use both 2018UL DoubleMuon and SingleMuon dataset). Since columnflow works in batches and in parallel it is no straightforward to check whether the "deterministic_seed" of an event is there twice. What would be the best approach to do this filtering in columnflow / has this been done in other analyses already?

**Comment by pkausw** [comment]
Hi @JulesVandenbroeck ,

I don't think there is a centralized way to handle your use case within columnflow at the moment. There are a couple of example in some analyses though:

- For the recent PO&DAS, we have set up a rather simple trigger logic. You can find the corresponding [selector](https://gitlab.cern.ch/cms-podas23/pag/cms-podas23-hig/-/blob/master/h4l/selection/trigger.py?ref_type=heads) and [config entries](https://gitlab.cern.ch/cms-podas23/pag/cms-podas23-hig/-/blob/master/h4l/config/config_podas.py?ref_type=heads#L34-L71) in the PO&DAS gitlab. If there are some issue with the access to this example, please let us know - there have been cases where people couldn't access these examples in the past, and the organizers have provided a centralized tar ball containing all examples used in the school
- The is a more involved example used e.g. in the [hh2bbtautau](https://github.com/uhh-cms/hh2bbtautau/tree/master) analysis, where dedicated Trigger objects are used. You can find these realized in [this config snippet](https://github.com/uhh-cms/hh2bbtautau/blob/master/hbt/config/triggers.py) and [this selector](https://github.com/uhh-cms/hh2bbtautau/blob/master/hbt/selection/trigger.py).

Both of these examples do not compare the `deterministic_seed`, but rather whether other triggers have already fired for an event to avoid the double counting.

Let us know if you have any additional questions!

Cheers,
Philip

**Reply by JulesVandenbroeck** [reply]
Thank you for your answer,

The trigger logic is indeed a convenient method to target double counting . This trigger logic has the down side that it is quite analysis specific so could not be added as a general columnflow feature which I was first thinking of (as removing double counting is important for every analysis). The comparison of `deterministic_seed` does not depend on the analysis so this would be the most useful, however it would need to be implemented after merging the data (I think) which is after the `cf.SelectEvents` task. I am thus a bit puzzled if this double counting removal can be done in a non analysis specific way without having to loop over all data events before selection.
At this point it is more a hypothetical "is this possible" question since the trigger logic is the more convenient probably.

Kind regards,
Jules
